src/terminal/char_grid/styles.rs

```diff
@@ -29,9 +29,25 @@ pub struct Styles {
     pub strikethrough: bool,
     pub inverted: bool,
     pub blink: bool,
+    config: Config,
```

Comment:
Sounds good. There'll be challenges to storing `Config` on `Terminal` because of the way `Deref` is being used to dispatch methods to `CharGrid`.